astropy/coordinates/distances.py

```diff
@@ -119,13 +123,6 @@ def __new__(cls, value=None, unit=None, z=None, cosmology=None,
                 raise ValueError('A `cosmology` was given but `z` was not '
```

Comment:
Refactoring to remove a level of indent might be a good idea, but it would clutter the diff and make it more difficult to review the changes that do change code behavior.

Reply:
I think the diff would be manageable. Either way.

Reply:
I see more opportunities for refactoring, so I'll leave it for a separate pull request.

Reply:
OK, sounds good.  And always good to not let the perfect be the enemy of the good